pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/AtcpAssentV2.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    private void revisionConsent(Handle handle, long adminUserId, StudyDto studyDto,
+                                 String activityCode, String versionTag, long timestamp) {
+        String reason = String.format(
+                "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                studyDto.getGuid(), activityCode, versionTag);
+        RevisionMetadata meta = new RevisionMetadata(timestamp, adminUserId, reason);
+
+        long activityId = ActivityBuilder.findActivityId(handle, studyDto.getId(), activityCode);
+        ActivityVersionDto version2 = handle.attach(ActivityDao.class).changeVersion(activityId, versionTag, meta);
+
+        updateVariables(handle, meta, version2);
+    }
+
+    private void updateVariables(Handle handle, RevisionMetadata meta, ActivityVersionDto version2) {
+        List<? extends Config> configList = dataCfg.getConfigList(TRANSLATION_UPDATES);
+        for (Config config : configList) {
+            revisionVariableTranslation(config.getString(TRANSLATION_KEY), config.getString(TRANSLATION_NEW),
+                    config.getString(LANGUAGE_CODE), handle, meta, version2);
+        }
+    }
+
+    private void revisionVariableTranslation(String varName, String newTemplateText, String isoLanguageCode, Handle handle,
+                                             RevisionMetadata meta, ActivityVersionDto version2) {
+        long tmplVarId = handle.attach(SqlHelper.class).findTemplateVariableIdByVariableName(varName);
+        JdbiVariableSubstitution jdbiVarSubst = handle.attach(JdbiVariableSubstitution.class);
+        List<Translation> transList = jdbiVarSubst.fetchSubstitutionsForTemplateVariable(tmplVarId);
+        List<Long> revisionIdList = transList.stream().map(Translation::getRevisionId).map(Optional::get).collect(Collectors.toList());
+        JdbiRevision jdbiRevision = handle.attach(JdbiRevision.class);
+        long newSubRevId = jdbiRevision.copyAndTerminate(revisionIdList.get(0), meta.getTimestamp(), meta.getUserId(), meta.getReason());
+        long[] retiringId = {newSubRevId};
+        transList.forEach(transListEntry -> {
+            jdbiVarSubst.bulkUpdateRevisionIdsBySubIds(Arrays.asList(transListEntry.getId().get()), retiringId);
+            if (isoLanguageCode.equalsIgnoreCase(transListEntry.getLanguageCode())) {
+                jdbiVarSubst.insert(transListEntry.getLanguageCode(), newTemplateText, version2.getRevId(), tmplVarId);
+            } else {
+                jdbiVarSubst.insert(transListEntry.getLanguageCode(), transListEntry.getText(), version2.getRevId(), tmplVarId);
+            }
+        });
+    }
+
+    private interface SqlHelper extends SqlObject {
+        @SqlQuery("select template_variable_id from template_variable where variable_name = :variable_name "
+                + "order by template_variable_id desc")
+        long findTemplateVariableIdByVariableName(@Bind("variable_name") String variableName);
```

Comment:
otchet-broad  
for future ref: In some cases, this might run into problem in dev where study was invalidated and reloaded. I meant we might get more results when queried by variable Name. Looks like here in dev we are getting only  1 row. We are good.